libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.ts

```diff
@@ -212,212 +243,184 @@ export class SkyDateRangePickerComponent
-    SkyDateRangeCalculatorId.NextCalendarYear,
-    SkyDateRangeCalculatorId.LastFiscalYear,
-    SkyDateRangeCalculatorId.ThisFiscalYear,
-    SkyDateRangeCalculatorId.NextFiscalYear,
-  ];
-  #_dateFormat: string | undefined;
-  #_disabled: boolean | undefined = false;
-  #_valueOrDefault: SkyDateRangeCalculation | undefined;
-
-  #changeDetector: ChangeDetectorRef;
-  #dateRangeService: SkyDateRangeService;
-  #formBuilder: UntypedFormBuilder;
-  #localeProvider: SkyAppLocaleProvider;
-  #ngZone: NgZone;
-
-  constructor(
-    changeDetector: ChangeDetectorRef,
-    dateRangeService: SkyDateRangeService,
-    formBuilder: UntypedFormBuilder,
-    localeProvider: SkyAppLocaleProvider,
-    ngZone: NgZone,
-    @Optional() themeSvc?: SkyThemeService,
-  ) {
-    this.#changeDetector = changeDetector;
-    this.#dateRangeService = dateRangeService;
-    this.#formBuilder = formBuilder;
-    this.#localeProvider = localeProvider;
-    this.#ngZone = ngZone;
-
-    this.dateFormatOrDefault = this.dateFormat;
-
-    this.#localeProvider
-      .getLocaleInfo()
-      .pipe(takeUntil(this.#ngUnsubscribe))
-      .subscribe((localeInfo) => {
-        SkyDateFormatter.setLocale(localeInfo.locale);
-        this.#preferredShortDateFormat =
-          SkyDateFormatter.getPreferredShortDateFormat();
-        this.dateFormatOrDefault =
-          this.dateFormat || this.#preferredShortDateFormat;
+  public ngAfterViewInit(): void {
+    this.#hostControl = this.#injector.get(NgControl, null, {
+      optional: true,
+      self: true,
+    })?.control;
+
+    // Set a default value on the control if it's undefined on init.
+    // We need to use setTimeout to avoid interfering with the first
+    // validation cycle.
+    if (isPartialValue(this.#hostControl?.value)) {
+      setTimeout(() => {
+        this.#hostControl?.setValue(this.#getValue(), {
+          emitEvent: false,
+          onlySelf: true,
+        });
       });
+    }
 
-    // Update icons when theme changes.
-    /* istanbul ignore next */
-    themeSvc?.settingsChange
-      .pipe(takeUntil(this.#ngUnsubscribe))
+    // Update the view when "required" or "disabled" states are changed on the
+    // host control.
+    this.#hostControl?.statusChanges
+      .pipe(distinctUntilChanged(), takeUntil(this.#ngUnsubscribe))
       .subscribe(() => {
         this.#changeDetector.markForCheck();
       });
-  }
-
-  public ngOnInit(): void {
-    this.#createForm();
-
-    this.#updateCalculators().then(() => {
-      if (!this.#value || !this.#value.calculatorId) {
-        this.#valueOrDefault = this.#defaultValue;
-      }
 
-      this.#addEventListeners();
-
-      this.isReady = true;
-
-      this.#showRelevantFormFields();
-
-      // We need to let Angular be stable and have rendered the components prior to setting the values and form controls. This ensures all initial validation will be ran correctly.
-      this.#ngZone.onStable.pipe(first()).subscribe(() => {
```

Comment:
I found some obscure articles online that mention using `ngZone` instead of `setTimeout`. Most people recommended using `setTimeout` since they essentially do the same thing (and one isn't an observable).